expression/builtin_string_vec.go

```diff
@@ -789,11 +789,49 @@ func (b *builtinFieldRealSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
 }
 
 func (b *builtinOctStringSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinOctStringSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		negative, overflow := false, false
+		str := buf.GetString(i)
+		str = getValidPrefix(strings.TrimSpace(str), 10)
+		if len(str) == 0 {
+			result.AppendString("0")
```

Comment:
Should we add a `continue` statement below this line?

Reply:
I will fix it.